docs/developers/developer-reference/sc-contract-calls.md

```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
 ```rust
-#[proxy]
-fn contract_proxy(&self, sc_address: ManagedAddress) -> callee_proxy::Proxy<Self::Api>;
+self.contract_proxy()
+	.init(my_biguint_arg)
 ```
 
+---
+
 [comment]: # (mx-context-auto)
 
-### Method #3: Manual calls (NOT recommended)
+## Contract deploy: configuration
 
-If for some reason you don't want to use the contract proxies, you can create a `ContractCall` object by hand:
+Just like with regular contract calls, we can specify the gas limit and perform EGLD transfers as part of the deploy as follows:
 
 ```rust
-let mut contract_call = ContractCall::new(
-    self.api,
-    dest_sc_address,
-    ManagedBuffer::new_from_bytes(endpoint_name),
-);
+self.contract_proxy(callee_sc_address)
+	.init(my_biguint_arg)
+
 ```
 
-Where `dest_sc_address` is the address of the callee contract, and `endpoint_name` would be `b"myEndpoint"`.
+```rust
+self.contract_proxy()
+	.init(my_biguint_arg)
+	.with_egld_transfer(egld_amount)
+    .with_gas_limit(gas_limit)
+```
 
-From here, you would use `contract_call.push_endpoint_arg(&your_arg)` and the manual payment adder functions described in method #1 miscellaneous category.  
+---
 
-You would then use the same `execute_on_dest_context`, `transfer_execute` and `async_call` methods as in the automatically built `ContractCall` objects from before.
+[comment]: # (mx-context-auto)
+
+## Contract deploy: execution
 
-Only use this method if you REALLY have to, as it's very easy to make mistakes if you work with low-level code like this.
 
 [comment]: # (mx-context-auto)
 
-## Conclusion
+### Deploy
+
+
+There are several ways to launch a contract deploy, different from a regular contract call.
+
+The simplest deploy operation we can perform is simply calling `deploy_contract`:
+
+
+```rust
+let (new_address, result)  = self.contract_proxy()
+	.init(my_biguint_arg)
+	.deploy_contract::<ResultType>(code, code_metadata);
+```
+
+:::important
+Contract deploys always happen in the same shard as the deployer. They are therefore always [synchronous calls](#synchronous-calls) and we get the result right away. Just like for `execute_on_dest_context` we need to either write a result with an explicit result type, or give the result type as type argument.
+
+Contract upgrades, on the other hand, can be sent to a different shard, and are therefore in essence asynchronous calls.
+:::
+
+The methods for executing contract deploys are as follows:
+- `.deploy_contract(code, code_metadata)` - deploys a new contract with the code given by the contract.
+- `.deploy_from_source(source_address, code_metadata)` - deploys a new contract with the same code as the code of the contract at `source_address`. The advantage is that the contract doesn't need to handle the new contract code, which could be quite a large data blob. This saves gas. It requires that we have the code already deployed somewhere else.
+
+
+[comment]: # (mx-context-auto)
+
+### Upgrade
```

Comment:
In the next mainnet release we'll have deffered inits for upgrades. We'll have to care for this as well very soon. Maybe I more places.

Reply:
I know ....
But first we need the implementation, then the docs ....